### PR TITLE
Introduce Contributing doc

### PR DESCRIPTION
**Pull request by wldeh**
Adds `CONTRIBUTING.md` guide to streamline new contributors.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+However, please keep in mind that maintainers will have to support the resulting code of the project,
+so do familiarize yourself with the following guidelines.
+
+* If you make any code changes:
+    * Follow the [Kotlin Coding Conventions](https://kotlinlang.org/docs/reference/coding-conventions.html).
+        * Use 4 spaces for indentation.
+        * Use imports with '*'.
+    * [Build the project](#building) to make sure it all works and passes the tests.
+* If you fix a bug:
+    * Write the test the reproduces the bug.
```
Comment (qurbonzoda) [inline CONTRIBUTING.md:18]:
```suggestion
    * Write the test that reproduces the bug.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Contributing Guidelines
+
+There are two main ways to contribute to the project &mdash; submitting issues and submitting 
+fixes/changes/improvements via pull requests.
+
+## Submitting changes
+
+Submit pull requests [here](https://github.com/Kotlin/kotlinx-io/pulls).
```
Comment (qurbonzoda) [inline CONTRIBUTING.md:8]: Could you please change this url?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# Contributing Guidelines
+
+There are two main ways to contribute to the project &mdash; submitting issues and submitting
```
Comment (qurbonzoda) [inline CONTRIBUTING.md:3]: The "submitting issues" part is missing in this document. Please add it as well.

**Review by qurbonzoda — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (wldeh) [reply CONTRIBUTING.md]: Yes, thank you for pointing this out.

File: CONTRIBUTING.md (hunk 3 above)
Comment (wldeh) [reply CONTRIBUTING.md:3]: Noted. I have dropped the `submitting changes` section for a `submitting prs` and `submit issues` section in the latest commit. For context, I initially had it in a similar format as https://github.com/Kotlin/kotlinx-io/blob/master/CONTRIBUTING.md.

**Pull request merged**